src/editor/ExpirationPanel.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { ToggleControl, DatePicker } from '@wordpress/components';
+import { isInTheFuture } from '@wordpress/date';
+import { useEffect, useState } from '@wordpress/element';
+
+const ExpirationPanel = ( {
+	expiration_date = null,
+	postStatus,
+	onMetaFieldChange,
+	createNotice,
+	removeNotice,
+} ) => {
+	const [ noticeId, setNoticeId ] = useState( false );
+	useEffect( () => {
+		if ( expiration_date && ! isInTheFuture( expiration_date ) ) {
+			if ( noticeId ) {
+				return;
+			}
+			createNotice(
+				'warning',
+				sprintf(
+					/* translators: %s is the expiration date */
+					__(
+						'This prompt has expired on %s and has been reverted to draft. Publishing it will reset the expiration date.',
+						'newspack-plugin'
+					),
+					new Date( expiration_date ).toLocaleDateString()
+				),
+				{
+					isDismissible: false,
+				}
+			).then( ( { notice } ) => {
+				setNoticeId( notice.id );
+			} );
+		}
+	}, [ expiration_date ] );
+
+	useEffect( () => {
+		if ( postStatus === 'publish' && noticeId ) {
+			removeNotice( noticeId );
+			createNotice(
+				'info',
+				__(
+					'This prompt has been published. The expiration date has been removed.',
+					'newspack-plugin'
+				)
+			);
+			// This is just for quicked feedback, the actual meta field deletion will
+			// happen on the backend.
+			onMetaFieldChange( { expiration_date: null } );
+		}
+	}, [ postStatus ] );
+
+	return (
+		<>
+			<ToggleControl
+				label={ __( 'Expiration Date', 'newspack-newsletters' ) }
+				checked={ !! expiration_date }
+				onChange={ () => {
+					if ( expiration_date ) {
+						onMetaFieldChange( { expiration_date: null } );
+					} else {
+						onMetaFieldChange( { expiration_date: new Date() } );
```

Comment:
Using the current date causes some strange behavior where toggling the expiration date feature on for the first time will trigger a notice that the prompt has expired and has been reverted to draft while it has not really been reverted to draft:
![Screenshot 2024-05-22 at 13 37 33](https://github.com/Automattic/newspack-popups/assets/17905991/87e53846-45f3-4f77-9783-6bc61e8b3a20)

Maybe we can make the default 24 hours in the future?

Reply:
Good catch! Fixed in 5cf76483d8991112d343b00cb6af66c9913950ec